### PR TITLE
HW01: `sendInLoop` detector [Mikhail Vorobev] 

### LINKED ISSUE #12: `send` in Loop
The detector should highlight loop statements that use the [`send`](https://docs.tact-lang.org/book/send#send-message) function call (`AstStaticCall`) in their bodies.

## Context
The `send` operation costs gas, i.e., it uses resources defined in a contract. Loop statements containing the `send` function call might be problematic, especially if the loop condition is unbounded or depends on user input or previous interactions with the contract. This operation might drain gas from the contract, effectively making it unusable for a while.

Thus, we want to highlight these cases for an auditor to review the relevant loop conditions.

## Example
```
while (a > self.num) {
  // other statements
  send(/* ... */); // Highlighted: Using send in the loop
}
```

---

The real-world implementation should be more complicated an use inter-procedural analysis: https://github.com/nowarp/misti/issues/131. We don't need this in the assignment.

### PR DESCRIPTION
**Pull request by InversionSpaces**
closes #12

### CODE REVIEW

File: assignments/1-ast-detectors/sendInLoop/sendInLoop.ts [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+  AstExpression,
+  AstNode,
+  AstStatement,
+  AstStatementForEach,
+  AstStatementRepeat,
+  AstStatementUntil,
+  AstStatementWhile,
+} from "@tact-lang/compiler/dist/grammar/ast";
+
+namespace AstHelpers {
```
Comment (jubnzv) [inline assignments/1-ast-detectors/sendInLoop/sendInLoop.ts:21]: We can either suppress it or create a utility file/module in the parent directory that is available to other detectors.

**Review by jubnzv — COMMENTED** [review]
LGTM, let's make ESLint happy before merging it.

File: assignments/1-ast-detectors/sendInLoop/sendInLoop.ts (hunk 1 above)
Comment (InversionSpaces) [reply assignments/1-ast-detectors/sendInLoop/sendInLoop.ts]: Moved this code to utils folder

Comment (jubnzv) [reply assignments/1-ast-detectors/sendInLoop/sendInLoop.ts]: Looks great, thank you

**Pull request merged**